### PR TITLE
Update the Playwright version

### PR DESCRIPTION
**Pull request by obulat**
## Description

This PR updates the Playwright version a newer one, 1.24. This will allow us to use HAR recordings introduced in version 1.23.
We pin the version to prevent accidental test failure due to changes in browser versions, as is advised in Playwright docs.

There were some problems with visual regression tests for RTL in version 1.25, see comments for more details.

## Testing Instructions

Playwright CI tests should pass.

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by zackkrida** [comment]
Here's another cool feature added in 1.24:

https://playwright.dev/docs/release-notes#-component-tests-update

**Comment by obulat** [comment]
I don't know what's happening, but the `node_modules` seems to be not available in the Docker container. This prevents the container from running Playwright and talkback together. I wonder if it's related to the multistage build.

**Comment by sarayourfriend** [comment]
We don't use a multi-stage Dockerfile for playwright. We also don't explicitly copy `node_modules` into the container because we expect it to be mounted by docker-compose: https://github.com/WordPress/openverse-frontend/blob/c4a117a11ddda2d0085506c88284db43e2784223/docker-compose.playwright.yml#L10

Indeed, it does get properly mounted:

```
➜  openverse-frontend git:(update/vue) ✗ USER_ID=$(id -u) docker-compose -f docker-compose.playwright.yml run playwright bash
pwuser@72718588e85e:/app# ls
CODE_OF_CONDUCT.md     LICENSE                        ecosystem.config.js      nuxt.config.ts      tailwind.config.js     typings
CONTRIBUTING.md        README.md                      feat                     package.json        tailwind.safelist.txt
DEPLOYMENT.md          TESTING_GUIDELINES.md          jest.config.js           pnpm-lock.yaml      test
Dockerfile             bin                            node_modules             prettier.config.js  test-results
Dockerfile.playwright  docker-compose.playwright.yml  nuxt-template-overrides  src                 tsconfig.json
pwuser@72718588e85e:/app# ls node_modules/.bin
acorn                     dot-object              import-local-fixture  nanoid           run-s                      ts-node-transpile-only
ansi-html                 editorconfig            is-ci                 nopt             sane                       ts-script
ansi-to-html              errno                   is-docker             npm-run-all      semver                     tsc
atob                      escodegen               jest                  nuxt             sentry-cli                 tsserver
autoprefixer              esgenerate              jest-runtime          nuxt-cli         sha.js                     uglifyjs
babylon                   eslint                  jiti                  nuxt-storybook   sitemap                    uuid
browserslist              eslint-config-prettier  js-beautify           nuxt-telemetry   start-storybook            vue-demi-fix
build-storybook           esparse                 js-yaml               opencollective   storybook-server           vue-demi-switch
color-support             esvalidate              jscodeshift           opener           svgo                       vue-i18n-extract
css-beautify              flat                    jsesc                 parser           tailwind                   vue-tsc
css-blank-pseudo          glob-all                json5                 pidtree          tailwind-config-viewer     watch
css-has-pseudo            handlebars              lint-staged           playwright       tailwindcss                webpack
css-prefers-color-scheme  he                      loose-envify          prettier         tailwindcss-config-viewer  webpack-bundle-analyzer
cssesc                    html-beautify           lz-string             regexp-tree      terser                     which
default-browser-id        html-minifier           miller-rabin          regjsparser      ts-node                    x-default-browser
detect                    html-minifier-terser    mime                  replace-in-file  ts-node-cwd
detect-port               husky                   mkdirp                rimraf           ts-node-esm
detective                 image-size              mustache              run-p            ts-node-script
```

Even `npm-run-all` is there and has the correct permissions (owned by `pwuser`):

```
pwuser@72718588e85e:/app# ll ./node_modules/.bin | grep npm-run-all
-rwxr-xr-x  1 pwuser pwuser   556 Aug 24 17:08 npm-run-all*
```

I can even execute it just fine.

It turns out that Playwright at some point changed the execution context of the webserver command to be the same directory as the configuration file instead of the cwd from where Playwright was called.

Adding `cwd: '/app',` to the `webServer` configuration fixes the issue.

**Comment by obulat** [comment]
> It turns out that Playwright at some point changed the execution context of the webserver command to be the same directory as the configuration file instead of the cwd from where Playwright was called.

I found the PR that changed the context: https://github.com/microsoft/playwright/pull/13626
I [tried](https://github.com/WordPress/openverse-frontend/pull/1646/commits/c0bad4db3fa4b87aee8a51e29a50499016116721) simply adding the  `cwd: '/app'`, but it didn't work.

**Comment by sarayourfriend** [comment]
It looks like the tests did run, there just appear to be a lot of visual regression failures. Could those come down to the change in browser version that accompanies a playwright upgrade? Maybe there was something else I missed in the logs?

**Comment by obulat** [comment]
@sarayourfriend, it seems that all of the test failures here are due to some differences in the Arabic texts, or the way they are rendered. You can see some `NUL` symbols on this snapshot, for example:

image

**Comment by sarayourfriend** [comment]
Interesting... I wonder if the fonts changed in the container's base image or something :thinking:

**Comment by obulat** [comment]
I tried other versions, and all up to 1.25 work well. You can see version 1.24 pass the CI with all the same setting, and v1.25 has lots of visualization regression failures. 
One more thing I noticed in v1.25 Docker container logs is "Running playwright: config file not found" when running the tests. Although it must have found it to run the tests...

**Comment by obulat** [comment]
I think we can keep the version updated to 1.24 in this PR, and invesigate newer versions later, as it's not such a high priority now.

File: package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
     "generate": "nuxt generate",
     "start": "nuxt start",
     "prod": "pnpm build:only && pnpm start",
+    "prod:playwright": "pnpm i18n:copy-test-locales && pnpm build:only && pnpm start",
```
Comment (sarayourfriend) [inline package.json:28]: Nit: can shorten this to remove duplication.

```suggestion
    "prod:playwright": "pnpm i18n:copy-test-locales && pnpm prod",
```

**Review by sarayourfriend — APPROVED** [review]
LGTM. Agreed that we don't need to push this all the way to the most recent version immediately. Can you open an issue to investigate further the issues with going up to 1.25?

**Review by zackkrida — APPROVED** [review]
Happy to see this land once the conflicts are fixed. LGTM.
